fixham/internal/version/version.go

```diff
@@ -0,0 +1,79 @@
+package version
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/Masterminds/semver/v3"
+	"golang.org/x/text/cases"
+	"golang.org/x/text/language"
+
+	"github.com/projectcalico/calico/fixham/internal/calico"
+)
+
+type Version string
+
+func (v *Version) Set(value string) error {
+	_v, err := semver.NewVersion(strings.TrimPrefix(value, "v"))
+	if err != nil {
+		return err
+	}
+	*v = Version(_v.String())
+	return nil
+}
+
+func (v *Version) String() string {
+	return string(*v)
+}
+
+func (v *Version) FormattedString() string {
+	if *v == "" {
+		return ""
+	}
+	return fmt.Sprintf("v%v", *v)
+}
+
+func (v *Version) Milestone() string {
+	_v := semver.MustParse(string(*v))
+	return fmt.Sprintf("%s v%d.%d.%d", cases.Title(language.English).String(calico.ProductName), _v.Major(), _v.Minor(), _v.Patch())
+}
+
+func (v *Version) Stream() string {
+	_v := semver.MustParse(string(*v))
+	return fmt.Sprintf("v%d.%d", _v.Major(), _v.Minor())
+}
+
+func (v *Version) ReleaseBranch(releaseBranchPrefix string) string {
+	branch := fmt.Sprintf("%s-%s", releaseBranchPrefix, v.Stream())
+	prerelese := semver.MustParse(string(*v)).Prerelease()
+	if prerelese != "" {
+		if strings.HasPrefix(prerelese, "1.") {
+			branch += "-1"
+		}
+	}
+	return branch
+}
+
+func (v *Version) NextVersion() Version {
+	_v := semver.MustParse(string(*v))
+	if _v.Prerelease() != "" && strings.HasPrefix(_v.Prerelease(), "1.") {
+		prerelease := semver.MustParse(_v.Prerelease())
+		nextPrerelease := prerelease.IncMajor()
+		vNextPrerelease := fmt.Sprintf("%v.%v", nextPrerelease.Major(), nextPrerelease.Minor())
+		vNext := semver.New(_v.Major(), _v.Minor(), _v.Patch(), vNextPrerelease, "")
+		return Version(vNext.String())
+	}
+	return Version(_v.IncMinor().String())
+}
+
+func NextBranch(currentVersion string, devTagSuffix, releaseBranchPrefix string) string {
```

Comment:
This function, as well as NextVersion, seem to be unused. Is that intentional?

Reply:
straglers from branch cutting.
They will be used in release as we do need to tag a blank commit after release for next version
Also with cutting new branch.

Reply:
@radTuti What I meant by this is that we shouldn't be commit any unused code as part of this PR - even if we think it will be used a in future commit. It's generally bad practice to commit "dead" code as part of a PR, so please remove any functions that are not called in this PR itself, so we can merge a minimally viable change set.

Any functions that are likely to be used in a future PR can be stored in another branch for now so that they are not lost, but it adds extra work for the reviewer when a PR has unused code in it.